.github/workflows/ci.yml

```diff
@@ -0,0 +1,106 @@
+name: Rust CI
+
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+  schedule:
+    - cron: '11 7 * * 1,4'
+
+# env:
+#   RUSTFLAGS: -Dwarnings
+#   RUSTDOCFLAGS: -Dwarnings
+
+jobs:
+
+  fmt:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          profile: minimal
+          components: rustfmt
+          override: true
+      - uses: actions-rs/cargo@v1
+        with:
+          command: fmt
+          args: --all -- --check
+
+  # clippy:
+  #   runs-on: ubuntu-latest
+  #   steps:
+  #     - uses: actions/checkout@v1
+  #     - uses: actions-rs/toolchain@v1
+  #       with:
+  #           toolchain: stable
+  #           profile: minimal
+  #           components: clippy
+  #           override: true
+  #     - uses: actions-rs/clippy-check@v1
+  #       with:
+  #         token: ${{ secrets.GITHUB_TOKEN }}
+  #         args: --workspace --tests --examples
+
+  docs:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@master
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: nightly
+          profile: minimal
+          components: rust-docs
+          override: true
+      - uses: swatinem/rust-cache@v1
+        with:
+          key: ${{ github.job }}
+      - uses: actions-rs/cargo@v1
+        with:
+          command: doc
```

Comment:
If this isn't going to also run the doc upload script like the Travis config does then it's not really worth running this. I'm fine with dropping that entirely, I don't look at the nightly docs and I doubt anyone else does. docs.rs exists for releases and it's simple enough to run `cargo doc` locally for master.

Reply:
Oh, I didn't realise there's a travis config :)  I probably agree that nightly docs aren't that important, I also didn't know they existed, I imagine most people use cargo doc for this.

The intention of this step is to ensure there are no bad links or formatting issues in the doc comments, together with the `RUSTDOCFLAGS=-Dwarnings` env this will fail if there are any warnings in the docs building.